configure.self

```diff
@@ -217,6 +217,14 @@ const char *foo (void) {
 void foo (void) {
   pthread_setname_np(pthread_self(), "abc");
 }
+' || \
+    mkl_compile_check "pthread_setname_darwin" "HAVE_PTHREAD_SETNAME_DARWIN" disable CC "-lpthread" \
+'
+#include <pthread.h>
```

Comment:
this should have the DARWIN.. define too

Reply:
Sure, done. I put the define in the compiler flags for congruency with the -D_GNU_SOURCE in the GNU check above.

Reply:
Good thinking, but we actually want to keep the command line argument defines to a minimum since they affect all files, exposing a wider, possibly incompatible API, where it is not desired.
So I suggest adding the ifdef-define-endif before the includes in tinycthread_extra.c

Reply:
I'm not sure I follow? I meant I put `-D_DARWIN_C_SOURCE` in the compiler flags for this `mkl_compile_check` invocation, rather than putting a `#define _DARWIN_C_SOURCE` in the actual source file. I don't think that has any bearing on where we define the `_DARWIN_C_SOURCE` macro for compilation of librdkafka itself.

> So I suggest adding the ifdef-define-endif before the includes in tinycthread_extra.c

I think you mean moving the `#define _DARWIN_C_SOURCE` from rd.h to tinycthread_extra.c? That makes sense to me, and so that's what I've done. Let me know if you meant something else.

Reply:
>  I don't think that has any bearing on where we define the _DARWIN_C_SOURCE macro for compilation of librdkafka itself.

Any compiler flags you pass to mkl_compile_check for a passing check will be added to the final compiler flags for the librdkafka build.

Reply:
> Any compiler flags you pass to mkl_compile_check for a passing check will be added to the final compiler flags for the librdkafka build.

I know you wrote mklove, so it's certainly possible that I'm wrong here, but I just don't see how that's true? The flags are `$5` on this line https://github.com/edenhill/librdkafka/blob/d979ad2eedfa9b276670dd4f3a0fa3d2505e36d6/mklove/modules/configure.base#L1332
and then `$5` isn't used elsewhere in the function, except to log the flags if the check fails. Are you perhaps thinking of `mkl_lib_check`, which does append its flags to LIBS if the check succeeds? 

Apologies if it seems like I'm being difficult! I just want to make sure this works the same way as the `-D_GNU_SOURCE` flag in the GNU check... and if `mkl_compile_check` works the way you suggest, then the GNU check is misbehaving and setting `-D_GNU_SOURCE` globally when it shouldn't.

Reply:
You are right! I had it confused with mkl_lib_check.
Current solution is fine :+1: